juju/juju
# [JUJU-2284] Reject charm store charms from Deploy and SetCharms, Upload on the controller side.

Reject charms without a source of charm-hub or local from Deploy and SetCharm, as well as when Upload is called. Initial part of removing charmstore charm support from the controller. Biggest change to the unit tests using cs charms.

## QA steps
No change to charm hub or local deploy
```sh
(cd tests; ./main.sh deploy test_deploy_charms)
```

Deploy of a charmstore charm should fail. The error message will improve with the completion of [JUJU-2230](https://warthogs.atlassian.net/browse/JUJU-2230).
```sh
$ juju deploy cs:ubuntu csubuntu
WARNING Charm store charms, those with cs: before the charm name, will not be supported in juju 3.1.
	Migration of this model to a juju 3.1 controller will be prohibited.
Located charm "ubuntu" in charm-store, revision 21
Deploying "csubuntu" from charm-store charm "ubuntu", revision 21 in channel stable on ubuntu@20.04/stable
ERROR charm-store not a valid charm origin source
```

## Documentation changes

(cd tests ; ./main.sh -v resources)
